src/Components/Server/src/Circuits/RemoteJSRuntime.cs

```diff
@@ -3,19 +3,40 @@
 
 using System;
 using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.Options;
 using Microsoft.JSInterop;
 
 namespace Microsoft.AspNetCore.Components.Server.Circuits
 {
     internal class RemoteJSRuntime : JSRuntimeBase
     {
+        private readonly CircuitOptions _options;
         private CircuitClientProxy _clientProxy;
 
+        public RemoteJSRuntime(IOptions<CircuitOptions> options)
+        {
+            _options = options.Value;
+        }
+
         internal void Initialize(CircuitClientProxy clientProxy)
         {
             _clientProxy = clientProxy ?? throw new ArgumentNullException(nameof(clientProxy));
         }
 
+        protected override object OnDotNetInvocationException(Exception exception, string assemblyName, string methodIdentifier)
+        {
+
+            if (_options.DotNetInteropExceptionDetails)
```

Comment:
Can we make it default to show detailed errors if `HostingEnvironment.IsDevelopment()`? That's what we do in `GenericWebHostedService`.

Reply:
We can set the default for this based on WebHostDefaults.DetailedErrorsKey, so I'll just add a configureoptions class to set this up based on the config value.

Reply:
I would like to get @davidfowl opinion before doing the environment.IsDevelopment thing as he and Damian had super strong feelings against it in the past.

Its being done here, but I'm not sure if specific constraint that forced that.
https://github.com/aspnet/AspNetCore/blame/master/src/Hosting/Hosting/src/GenericHost/GenericWebHostedService.cs#L154

It would be good that whether or not to show detailed errors came from a single source of data and with a single way to configure. I believe I remember that is one of the things Barry pushed hard in the past, as people will misconfigure these things and leave the app open to leaks.

So what I'm proposing is:
1) Use a single switch for all asp.net core for detailed errors
2) Have one place where the logic for that switch is and is shared across all aspnet core components that need to do something similar to this.

@davidfowl @SteveSandersonMS thoughts?

For right now, using WebHostDefaults.DetailedErrorsKey is good enough IMO

Reply:
> For right now, using WebHostDefaults.DetailedErrorsKey is good enough IMO

That sounds good to me.